src/main/kotlin/uk/co/poolefoundries/baldinggate/core/CoreComponents.kt

```diff
@@ -72,6 +73,21 @@ data class PositionComponent(val x: Int, val y: Int) : Component {
 
 }
 
+data class AnimationComponent(val entity: Entity, var positions: List<PositionComponent> = listOf<PositionComponent>(), var progress: Float = 0F, var isFinished: Boolean = false) : Component {
+    val animationDuration = 0.1F
+
+    fun animationStep(delta: Float) {
+        this.progress += delta / animationDuration
+        val positionIndex = round(this.progress).toInt()
+        if (positionIndex >= this.positions.size-1){
+            this.isFinished = true
+            entity.add(positions.last())
```

Comment:
I was imagining the animation being purely visual. Like the skeleton should have it's actual position component updated instantly and the animation is just a visual effect that overloads the position the sprite is rendered at.